### PR TITLE
[fix] gracefully handle no "updpkgsums" in packages hooks

### PR DESCRIPTION
**Pull request by jmelahman**
Without installed:
```
$ git commit -m 'foo'
[ buildifier-bin ] setting pkgrel to '1'
[ buildifier-bin ] updating package checksums... FAIL
Command 'updpkgsums' not found. Try installing 'pacman-contrib'.

```

With installed:
```
$ git commit -m 'foo'
[ buildifier-bin ] setting pkgrel to '1'
[ buildifier-bin ] updating package checksums... OKAY
[ buildifier-bin ] updating .SRCINFO... OKAY
[jamison/buildifier-6 cc2e0eb] foo
 3 files changed, 10 insertions(+), 6 deletions(-)

```

### CODE REVIEW

**Review by sudoforge — CHANGES_REQUESTED** [review]
Thanks for this. We should probably do the same with `makepkg` and `git`, too.

Do you mind abstracting this out into a loop that checks these three dependencies, prints out an error message for each that is missing, and then exits with `1` (if any were missing)?

**Comment by jmelahman** [comment]
> Thanks for this. We should probably do the same with `makepkg` and `git`, too.
> 
> Do you mind abstracting this out into a loop that checks these three dependencies, prints out an error message for each that is missing, and then exits with `1` (if any were missing)?

Done

File: .githooks.d/pre-commit/packages.sh [hunk 1]
```diff
@@ -2,6 +2,25 @@
 
 set -e
 
+_RUNTIME_DEPS=('updpkgsums' 'makepkg' 'git')
```
Comment (jmelahman) [inline .githooks.d/pre-commit/packages.sh:5]: FYI, I haven't tried to write POSIX compliant shell before. Shellcheck complains the following, but there are similar warnings throughout this script, so not sure if these are worrisome,
```
In ./.githooks.d/pre-commit/packages.sh line 5:
_RUNTIME_DEPS=('updpkgsums' 'makepkg' 'git')
              ^----------------------------^ SC3030 (warning): In POSIX sh, arrays are undefined.


In ./.githooks.d/pre-commit/packages.sh line 8:
  local missing_deps=()
  ^----------------^ SC3043 (warning): In POSIX sh, 'local' is undefined.
                     ^-- SC3030 (warning): In POSIX sh, arrays are undefined.


In ./.githooks.d/pre-commit/packages.sh line 9:
  for d in "${_RUNTIME_DEPS[@]}"; do
            ^-----------------^ SC3054 (warning): In POSIX sh, array references are undefined.


In ./.githooks.d/pre-commit/packages.sh line 11:
      missing_deps+=("$d")
      ^----------^ SC3024 (warning): In POSIX sh, += is undefined.
                    ^----^ SC3030 (warning): In POSIX sh, arrays are undefined.


In ./.githooks.d/pre-commit/packages.sh line 16:
    for d in "${missing_deps[@]}"; do
              ^----------------^ SC3054 (warning): In POSIX sh, array references are undefined.


In ./.githooks.d/pre-commit/packages.sh line 17:
      >&2 printf "Command '$d' not found.\n"
                 ^-------------------------^ SC2059 (info): Don't use variables in the printf format string. Use printf '..%s..' "$foo".


In ./.githooks.d/pre-commit/packages.sh line 19:
    >&2 printf "Hint: use 'pacman -F ${missing_deps[*]}' to determine which packages to install."
               ^-- SC2059 (info): Don't use variables in the printf format string. Use printf '..%s..' "$foo".
                                     ^----------------^ SC3054 (warning): In POSIX sh, array references are undefined.
```

Happy to address any issue, but as-is looks most familiar to me.

Comment (sudoforge) [reply .githooks.d/pre-commit/packages.sh:5]: `shellcheck` is indeed correct; those arrays you're used to using are supported in `bash`, but are not part of the POSIX standard.

a good ol' space-separated string (which you call without quotes; see below) is all we need.

```
DEPS="foo bar baz"

for dep in $DEPS; do
    ...
done
```

**Review by sudoforge — CHANGES_REQUESTED** [review]
thanks for abstracting this out into a function! i have a few recommendations for improving readability and formatting; feel free to apply these as-is or use as inspiration for refactoring it yourself.

File: .githooks.d/pre-commit/packages.sh [hunk 2]
```diff
@@ -2,6 +2,25 @@
...
+  if arg_set "$missing_deps"; then
+    >&2 printf "FAIL\n"
+    for d in ${missing_deps}; do
+      >&2 printf "Command '%s' not found.\n" "$d"
+    done
+    >&2 printf "Hint: use 'pacman -F %s' to determine which packages to install." "$missing_deps"
+    exit 1
+  fi
+}
+
```
[the remark this answers is not in the document]
Comment (jmelahman) [reply .githooks.d/pre-commit/packages.sh]: Thank you

**Comment by sudoforge** [comment]
sorry this never merged; i have moved on from maintaining these packages. i'll merge this in-tree, but will be archiving the repository shortly.

**Review by sudoforge — APPROVED** [review]

**Pull request merged**